core/src/main/kotlin/com/malinskiy/marathon/execution/device/DeviceActor.kt

```diff
@@ -70,10 +70,10 @@ class DeviceActor(private val devicePoolId: DevicePoolId,
         }
         state<DeviceState.Running> {
             on<DeviceEvent.Terminate> {
-                transitionTo(DeviceState.Terminated, DeviceAction.Terminate(testBatch))
+                transitionTo(DeviceState.Terminated, DeviceAction.Terminate(testBatch, result))
```

Comment:
We need to make sure this is covered with tests. I can't follow all the graph transitions here properly without tests.